safe/definitions/fields.py

```diff
@@ -2505,6 +2505,109 @@
     'units': currencies,
 }
 
+# # # # # # # # # #
+# Direction and Distance Field
+# # # # # # # # # #
+
+# Count for each exposure type
+distance_field = {
+    'key': 'distance_field',
+    'name': tr('Distance'),
+    'field_name': 'distance',
+    'header_name': tr('Distance'),
+    'type': QVariant.Double,
+    'length': default_field_length,
+    'precision': default_field_precision,
+    'absolute': True,
+    'description': tr(
+        'The distance value between place and hazard point.'
+    ),
+    'help_text': tr(
+        'Distance value between place feature to the epicenter of the hazard. '
+        'The distance is calculated using WGS84 as ellipsoid model. The unit of'
+        'the distance is in meter.'),
+    'citations': [
+        {
+            'text': None,
+            'link': None
+        }
+    ],
+    # Null value can be replaced by default or not
+    'replace_null': False
+}
+
+bearing_from_field = {
+    'key': 'bearing_from_field',
+    'name': tr('Bearing Angle'),
+    'field_name': 'bearing_fr',
+    'header_name': tr('Bearing'),
+    'type': QVariant.Double,
+    'length': default_field_length,
+    'precision': default_field_precision,
+    'absolute': False,
+    'description': tr(
+        'An azimuth angle at a place to a hazard point.'),
+    'help_text': tr(
+        'A bearing angle is an angle measured to a point as observed in current'
+        ' location using north as a reference direction. In this case, "bearing'
+        ' from" refers to an angle calculated at a certain place pointing to a '
+        'hazard location. Positive values indicate it calculates from North '
+        'moving clockwise, and negative values indicate it calculates from '
+        'North moving counterclockwise.'),
+    'citations': [
+        {
+            'text': None,
+            'link': None
+        }
+    ]
+}
+
+# TODO: Ivan to update
+direction_field = {
+    'key': 'direction_field',
+    'name': tr('Direction'),
+    'field_name': 'dir_from',
```

Comment:
Why not use a full name? Like `direction_from`. And why not only `direction`?

Reply:
AFAIK, there is a limitation on how many numbers of character a field name can have, which is around 10 character IIRC. 

The reason I also add `from` and not only leave it as `direction` because it is a relative value which depends on where our base point is. For example, if the azimuth from A to B is 90 degree, then the `direction_from` A to B would be East, and the `direction_to` A from B would be West.

**But**, since at the moment we only use one of them (`direction from`, to get the direction value from a hazard point to places points), I think I will change it to use `direction` and add some more information in the help text.